src/main/java/io/javalin/core/JavalinServletConfig.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2017 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin.core;
+
+import io.javalin.RequestLogger;
+import io.javalin.core.util.CorsUtil;
+import io.javalin.core.util.SinglePageHandler;
+import io.javalin.security.AccessManager;
+import io.javalin.security.SecurityUtil;
+import io.javalin.staticfiles.JettyResourceHandler;
+import io.javalin.staticfiles.Location;
+import io.javalin.staticfiles.ResourceHandler;
+import io.javalin.staticfiles.StaticFileConfig;
+
+// @formatter:off
+public class JavalinServletConfig {
+
+    public boolean dynamicGzip = true;
+    public boolean autogenerateEtags = false;
+    public boolean prefer405over404 = false;
+    public String defaultContentType = "text/plain";
+    public Long requestCacheSize = 4096L;
+    RequestLogger requestLogger;
+    ResourceHandler resourceHandler;
+    AccessManager accessManager = SecurityUtil::noopAccessManager;
+    SinglePageHandler singlePageHandler = new SinglePageHandler();
+    private JavalinServlet servlet;
+
+    public JavalinServletConfig(JavalinServlet servlet) {
+        this.servlet = servlet;
```

Comment:
I see we need to keep reference to the JavalinServlet for `enableCorsForOrigins(...)`. Should we (maybe at some point in the future) rework and execute those two handlers related to cors in `JavalinServlet.service()` with something like:
```java
public class JavalinServletConfig {
...
public Set<String> corsAllowOrigins;
...
}
// somewhere in JavalinServlet.service()
if (config.corsAllowOrigins is not empty) {
... do something
}
```

Imho, it would be easier to debug all configuration behavior in a single place (by now it's `servlet.service()` since you there are most of `config.isSomethingEnabled?` checks and relevant logic in there).  I think it's ok as it is, but worth considering.

Reply:
It doesn't feel like this is something that should belong to `JavalinServlet` at all, it's just a convenience method so people won't have to write their own before/options handlers 🤔
I'm not happy about how it works now (a Config class having a reference to what it's supposed to be a config for), but I would rather just move it to `app.enableCors...` again than deeper into `JavalinServlet`.

Reply:
I see your point here, since these are handlers which could be registered via `addHandler` but since `cors` is config option, this logic happens to appear in ServletConfig. I think at least there's point in this. Let's see if we could hold handlers in the config (without altering servlet itself) and add them all when applying configuration. For now it's out of scope of this task then :)